### ISSUE TITLE
Conversion error reading vlen string dataset

### ISSUE THREAD

**Issue by jreadey** [h5py/h5py#565]
This file contains one dataset with a vlen string type: https://github.com/HDFGroup/h5serv/blob/master/testfiles/vlen_string_dset_utc.h5. 

h5dump output looks normal, but if I open with h5py I get a conversion error:

```
f = h5py.File("vlen_dset_utc.h5", "r")
ds1 = f["ds1"]
print ds1[0]
```

Outputs:

```
KeyError: 'Conversion function not found (No appropriate function for conversion path)'
```

Any ideas what is wrong?

**Comment by tacaswell** [comment]
I do not remember the details off the top of my head, but I think this is related to numpy not having a good data type to map to vlen strings (as the 'native' string types are fixed length).  I do not recall the reason not to use 'object' as the dtype in this case though.

**Comment by andrewcollette** [comment]
We do support vlen strings using the 'O' type... the function h5py.special_dtype can be used to construct the appropriate type.

I get the same error, but weirdly I can create vlen arrays from h5py with no problem.  I can't find any difference between the data types in the two files.

```
import h5py
f = h5py.File("vlen.hdf5")
ds = f.create_dataset("x", (100,), dtype=h5py.special_dtype(vlen=str))
ds[0] = "Hello"
f.close()
```

**Comment by jreadey** [comment]
Right, compare with this file: https://github.com/HDFGroup/h5serv/blob/master/testfiles/vlen_string_nullterm_dset.h5.   Looks like exactly the same type and dataset attributes but it loads fine with h5py while the original one does not.

**Comment by jreadey** [comment]
This is the current path for the file illustrating the problem: https://github.com/HDFGroup/hdf5-json/blob/master/data/hdf5/vlen_string_dset_utc.h5.

**Comment by aragilar** [comment]
So the distinction seems to be spacepad vs nullterm between the files, I wonder if there some additional conversion happening along the line which lets spacepad work?

**Comment by takluyver** [comment]
We believe that this was fixed by #1608. If people can still replicate it after that, please either reopen this issue or create a new one.
